### PR TITLE
feat/voltron adapt v8

### PR DESCRIPTION
**Pull request by lvfen**
# Pre-PR Checklist

- [x] I added/updated relevant documentation.
- [x] I followed the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters to submit commit message.
- [x] I squashed the repeated code commits.
- [x] I signed the [CLA].
- [x] I added/updated test cases to check the change I am making.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=1944) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=1944) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: lvfen<br/>:x: henryjin<br/><hr/>**henryjin** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/Hippy?pullRequest=1944) it.</sub>

File: driver/js/flutter/core/src/bridge/android/bridge_impl.cc [hunk 1]
```diff
@@ -105,7 +105,7 @@ bool BridgeImpl::RunScriptFromFile(int64_t runtime_id,
                                    std::function<void(int64_t)> callback) {
   TDF_BASE_DLOG(INFO) << "RunScriptFromFile begin, runtime_id = "
                       << runtime_id;
-  std::shared_ptr<Runtime> runtime = Runtime::Find(runtime_id);
+  std::shared_ptr<Runtime> runtime = Runtime::Find(static_cast<int32_t>(runtime_id));
```
Comment (churchill-zhang) [inline driver/js/flutter/core/src/bridge/android/bridge_impl.cc:108]: 需要用宏校验下

File: driver/js/flutter/core/src/ffi/ffi_bridge_runtime.cc [hunk 2]
```diff
@@ -61,9 +63,9 @@ void FFIJSBridgeRuntime::ReportJSException(std::u16string &description_stream, s
 }
 
 void FFIJSBridgeRuntime::SendResponse(const uint16_t* source, int len) {
-  void* copy = (void*)malloc(len * sizeof(uint16_t));
-  memset(copy, 0, len * sizeof(uint16_t));
-  memcpy(copy, (void*)source, len * sizeof(uint16_t));
+  void* copy = (void*)malloc(static_cast<unsigned int>(len) * sizeof(uint16_t));
```
Comment (churchill-zhang) [inline driver/js/flutter/core/src/ffi/ffi_bridge_runtime.cc:66]: 干脆改成智能指针吧

File: driver/js/flutter/core/src/ffi/bridge_ffi_impl.cc [hunk 3]
```diff
@@ -148,7 +148,7 @@ EXTERN_C void CallFunctionFFI(int32_t engine_id, const char16_t* action, const c
                                  [callback_id](int64_t value) { CallGlobalCallback(callback_id, value); });
       } else {
         auto copy_params = voltron::CopyCharToChar(params, params_length);
```
Comment (churchill-zhang) [inline driver/js/flutter/core/src/ffi/bridge_ffi_impl.cc:150]: copyCharToChar也换智能指针吧

File: driver/js/flutter/android/gradle.properties [hunk 4]
```diff
@@ -2,8 +2,56 @@ org.gradle.jvmargs=-Xmx1536M
...
-#V8
-V8_TAG=recommend
-#V8_RELEASE=x5-lite
-#V8_RELEASE=official-release
-EXT_JS_ENGINE_LIB=false
+#
+# V8 Component
+#
+# The following prebuilt V8 component tag are available:
+#   * 9.7-lkgr (recommend)
```
Comment (churchill-zhang) [inline driver/js/flutter/android/gradle.properties:9]: 这个不复用外面的gradle吗

File: driver/js/core/src/runtime/v8/v8_bridge_utils.cc [hunk 5]
```diff
@@ -487,23 +487,28 @@ void V8BridgeUtils::CallJs(const unicode_string_view& action,
...
       hippy::napi::V8TryCatch try_catch(true, context);
-      v8::ValueDeserializer deserializer(
-          isolate, reinterpret_cast<const uint8_t*>(buffer_data_.c_str()),
-          buffer_data_.length());
-      TDF_BASE_CHECK(deserializer.ReadHeader(ctx).FromMaybe(false));
-      v8::MaybeLocal<v8::Value> ret = deserializer.ReadValue(ctx);
-      if (!ret.IsEmpty()) {
-        params = std::make_shared<hippy::napi::V8CtxValue>(
-            isolate, ret.ToLocalChecked());
+      v8::MaybeLocal<v8::Value> ret;
```
Comment (churchill-zhang) [inline driver/js/core/src/runtime/v8/v8_bridge_utils.cc:490]: 需要提前判空

File: driver/js/flutter/cmake/framework_platform_android.cmake [hunk 6]
```diff
@@ -5,16 +5,26 @@ set(ABI_COMPILE_OPTIONS
 add_compile_options(${ABI_COMPILE_OPTIONS})
 
 set(JS_ENGINE "V8")
-set(JNI_DIR ${FRAMEWORK_DIR}/js/android/src/main/jni)
-set(V8_COMPONENT "${JNI_DIR}/third_party/v8/stable/official-release")
 
 set(FRAMEWORK_ANDROID_DEPS android log)
```
Comment (churchill-zhang) [inline driver/js/flutter/cmake/framework_platform_android.cmake:9]: 感觉都可以复用其他模块对v8的处理

File: driver/js/flutter/android/gradle.properties (hunk 4 above)
Comment (lvfen) [reply driver/js/flutter/android/gradle.properties:9]: voltron内部的gradle配置还是有一些不一样的，不能完全复用

**Review by churchill-zhang — APPROVED** [review]

File: driver/js/flutter/cmake/framework_platform_android.cmake (hunk 6 above)
Comment (lvfen) [reply driver/js/flutter/cmake/framework_platform_android.cmake:9]: 后面抽出driver后切过去就好

**Review by medns — APPROVED** [review]
LGTM

**Pull request merged**